include/s3select.h

```diff
@@ -1100,6 +1116,42 @@ void push_data_type::operator()(s3select* self, const char* a, const char* b) co
   }
 }
 
+void push_trim_expr::operator()(s3select* self, const char* a, const char* b) const
+{
+  std::string token(a, b);
+  
+  std::string trim_function;
+
+  trim_function = self->getAction()->dataTypeQ.back();
```

Comment:
dataTypeQ is queue for cast operator.
its better to split it, to another separate queue.
in nested calls, (trim->cast , cast->trim) , it may cause wrong behaviour.